react-app-drawer/my-app/src/components/AppDrawer.tsx

```diff
@@ -0,0 +1,41 @@
+import { useState } from 'react';
+import { Outlet } from 'react-router-dom';
+import { FaBars } from 'react-icons/fa6';
+import { MenuLink } from './MenuLink';
+
+export type MenuItem = {
+  name: string;
+  iconUrl: string;
+  path: string;
+};
+
+type Props = {
+  menuItems: MenuItem[];
+};
+export function AppDrawer({ menuItems }: Props) {
+  const [isOpen, setIsOpen] = useState(true);
+  return (
+    <div className="flex w-full h-full">
+      <nav
+        className={`overflow-hidden transition-basis duration-500 ${
+          isOpen ? 'basis-[250px] ' : 'basis-[60px]'
+        } shadow-[5px_0_5px_#ccc]`}>
+        <FaBars
+          className="w-7 h-7 my-4 ml-1 mr-7 hover:cursor-pointer"
+          onClick={() => setIsOpen(!isOpen)}
+        />
+        {isOpen && (
+          <h2 className="m-2 text-[26px] font-semibold ">Hylian Shopping</h2>
+        )}
+        <ul>
```

Comment:
The "constraint" in the exercise was actually trying to get you to move the entire `<ul></ul>` block into a separate component, say, Menu, which maps through the items. That way, you would just render the Menu component here and pass it the entire list of menu items.

It's not a big deal, though. That constraint was added a while ago when I was worried about giving students enough experience working with cooperating components. The curriculum has been updated to where you get plenty of that experience, so the constraint isn't as vital as it was.